source/fab/tools/compiler.py

```diff
@@ -92,68 +93,95 @@ def check_available(self) -> bool:
             this by requesting the compiler version.
         '''
         try:
-            version = self.get_version()
-        except RuntimeError:
-            # Compiler does not exist:
+            self.get_version()
+            # A valid version means the compiler is available.
+            return True
+        except RuntimeError as err:
+            # Compiler does not exist, or version could not be handled:
+            self.logger.error(f'Error getting compiler version: {err}')
             return False
 
-        # An empty string is returned if some other error occurred when trying
-        # to get the compiler version.
-        return version != ""
-
```

Comment:
I'm not very familiar w/ this code - is there a chance an exception other than `RuntimeError` could happen?  In that case should you remove the return statement in the try block and re-insert the `version != ""` here?

Reply:
`get_version()` will no longer return an empty string, and there is no other 'empty' type value to check for. It will always raise a `RuntimeError` if it can't determine the version.
`RuntimeError` is the only expected error type from the underlying `self.run()` function (which is in `tool.py`). It handles the `FileNotFoundError` if the compiler isn't found at all.